R/expect_identical_linter.R

```diff
@@ -69,6 +69,11 @@ expect_identical_linter <- function() {
         and expr[NUM_CONST[contains(text(), '.')]]
       ]
       or following-sibling::expr[NUM_CONST[contains(text(), '.')]]
+      or following-sibling::expr[
+        OP-MINUS
+        and count(expr) = 1
+        and expr[NUM_CONST[contains(text(), '.')]]
```

Comment:
I think the last condition belongs in `[]` -- otherwise it can get hard to visually parse operator presence of `=` vs. nearby logical operators like `and`/`or`. IINM occasionally we have to let `/text()` "out" e.g. to compare to the `text()` of another node, but it's best avoided.

We don't really have an XPath style guide, so these are just my thoughts from writing many of these, you or other lintr authors may disagree.